packages/connector-tencent-email/src/index.ts

```diff
@@ -0,0 +1,150 @@
+import type {
+  CreateConnector,
+  EmailConnector,
+  GetConnectorConfig,
+  SendMessageFunction,
+} from '@logto/connector-kit';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  ConnectorType,
+  validateConfig,
+} from '@logto/connector-kit';
+import { assert } from '@silverhand/essentials';
+import { HTTPError } from 'got';
+
+import { isErrorResponse, request } from '@/http';
+import type { MailTemplateType, SendTencentMailConfig, ValueNames } from '@/types';
+import { sendTencentMailConfigGuard } from '@/types';
+
+import { defaultMetadata } from './constant';
+
+function getTemplate(templates: MailTemplateType[], usageType: string): MailTemplateType {
+  const template = templates.find((template) => template.usageType === usageType);
+  assert(
+    template,
+    new ConnectorError(
+      ConnectorErrorCodes.TemplateNotFound,
+      `Template not found for type: ${usageType}`
+    )
+  );
+
+  return template;
+}
+
+function buildFromName(fromName: string, fromAddress: string): string {
+  return fromName.length > 0 ? `${fromName} <${fromAddress}>` : fromAddress;
+}
+
+function buildTemplatePayload(
+  template: MailTemplateType,
+  templateParameters: Record<typeof ValueNames[number], string>
+): Record<string, string> {
+  return template.params.reduce<Record<string, string>>((accumulator, parameter) => {
+    const { name, value } = parameter;
+
+    return {
+      ...accumulator,
+      [name]: templateParameters[value],
+    };
+  }, {});
+}
```

Comment:
I actually don't quite get the purpose of this util function. Looks like you are just merging two records, am I right?

Reply:
Looks you can even do it with:
```ts
return { ...template, ...templateParameters }
```

Reply:
I might be wrong but it seems the only key the two objects have in common is `subject`, isn't it?

<img width="408" alt="image" src="https://user-images.githubusercontent.com/12833674/208701481-e78eb9cd-153f-4827-87ac-cc91262b1600.png">